api/app/Http/Controllers/UsersController.php

```diff
@@ -0,0 +1,31 @@
+<?php
+
+namespace App\Http\Controllers;
+
+use App\Models\User;
+use Illuminate\Http\JsonResponse;
+use Illuminate\Http\Request;
+use Illuminate\Support\Facades\Hash;
+
+class UsersController extends Controller
+{
+    public function register(Request $request): JsonResponse
+    {
+        $validated = $request->validate([
+            'username' => 'required|string|max:255',
+            'email' => 'required|string|email|max:255|unique:users',
+            'password' => 'required|string',
+        ]);
```

Comment:
@majamitic774 The frontend sends `password_confirmation` but it's not being validated here. This is a security issue. Should add password confirmation validation.

I'm not sure. But I think `confirmed` rule will automatically check for `password_confirmation` field.